lib/deprecation_tracker.rb

```diff
@@ -28,7 +28,11 @@ def warn(*messages, uplevel: nil, category: nil)
       elsif Gem::Version.new(RUBY_VERSION) < Gem::Version.new("3.0")
         super(*messages, uplevel: nil)
       else
-        super
+        begin
+          super(*messages, uplevel: uplevel, category: category, **kwargs)
+        rescue ArgumentError => e
```

Comment:
Some logging implementations (e.g., in native gems) may not accept arbitrary keyword args. This fallback ensures compatibility by only passing the known, expected keys.